### ISSUE TITLE
Exception when trying to use with recent HA version like 2023.09

### ISSUE THREAD

**Issue by thomluther** [walthowd/ha-constant-modifier#2]
Seems the last change to this is quite old and it causes configuration load exceptions for the home assistant object since it has now mandatory arguments as described here:
https://developers.home-assistant.io/blog/2023/08/29/homeassistant-changed-signature?_highlight=homeassistant%28

With following changes I could get the ha-constant-modifier to work with HA 2023.10:
Updated imports:
```
from homeassistant.core import HomeAssistant
from homeassistant.config import load_yaml_config_file, YAML_CONFIG_FILE
from homeassistant.__main__ import get_arguments
from homeassistant.util.yaml import SECRET_YAML, Secrets, load_yaml
```

here is the updated def get_ha_config() to use the required config file parameter and the secrets parameter.
PS: From the code it appears that Secret is optional and defaults to Null, but since I use Secrets includes in my  config I guess this must be specified as well when you want to duplicate the changed HA startup sequence

```
def get_ha_config():
    """
    Duplicate enough of the HA startup sequence to extract the config *really* early.
    """

    args = get_arguments()

    try:
        hass = HomeAssistant()  # pylint: disable=no-value-for-parameter
        hass.config.config_dir = os.path.abspath(os.path.join(os.getcwd(), args.config))
    except TypeError:
        hass = HomeAssistant(os.path.abspath(os.path.join(os.getcwd(), args.config)))  # pylint: disable=too-many-function-args

    return load_yaml_config_file(hass.config.path(YAML_CONFIG_FILE),Secrets(hass.config.path(hass.config.config_dir)))
```

**Comment by Pat28** [comment]
Anyone using this in 2025.2?

**Comment by puddly** [comment]
I don't think this will work in newer versions of HA, they load custom components very late.

**Comment by Pat28** [comment]
Ok, the reason I ask, is that I use the SolarEdge integration but only want it to update every hour as I get into 'too many request issues' as Domoticz is also interrogating that API. Any other workaround?

**Comment by thomluther** [comment]
It does not look like this repo is still maintained. I did a fork and changed it to recent HA versions. It still works for my ZHA Quirk constant changes. Not sure whether it still works for other constants that may have to be modified early in the load.
https://github.com/thomluther/ha-constant-modifier
Let me know if that works.

**Comment by puddly** [comment]
ZHA quirks are a little special because they're modules that are lazily loaded by ZHA well after it starts, it's sort of the only exception.

@thomluther Out of curiosity, what constants do you modify in the quirks module?

**Comment by thomluther** [comment]
Hi @puddly Same as described in the Repo:
https://github.com/thomluther/ha-constant-modifier#configuration
```yaml
constant_modifier:
  zhaquirks.xiaomi:
    MotionCluster.reset_s: 5 # default 70
    OccupancyCluster.reset_s: 180 # default 600
```

This requires also a HW hack on the Motion device, but it seems to be the only way to get shorter timeouts as lengthy discussed in HA community forum post [Xiaomi Human / Body / Motion Sensor - Timeout](https://community.home-assistant.io/t/xiaomi-human-body-motion-sensor-timeout/23398/481)

**Comment by Pat28** [comment]
Wow, thanks, this seems to work, at least it is reporting the values

2025-05-06 15:57:55.079 WARNING (MainThread)
[custom_components.constant_modifier] Patched
homeassistant.components.solaredge.const, OVERVIEW_UPDATE_DELAY =
timedelta(hours=1) (was 0:15:00)
2025-05-06 15:57:55.080 WARNING (MainThread)
[custom_components.constant_modifier] Patched
homeassistant.components.solaredge.const, DETAILS_UPDATE_DELAY =
timedelta(hours=12) (was 12:00:00)
2025-05-06 15:57:55.081 WARNING (MainThread)
[custom_components.constant_modifier] Patched
homeassistant.components.solaredge.const, INVENTORY_UPDATE_DELAY =
timedelta(hours=12) (was 12:00:00)
2025-05-06 15:57:55.082 WARNING (MainThread)
[custom_components.constant_modifier] Patched
homeassistant.components.solaredge.const, POWER_FLOW_UPDATE_DELAY =
timedelta(hours=1) (was 0:15:00)
2025-05-06 15:57:55.082 WARNING (MainThread)
[custom_components.constant_modifier] Patched
homeassistant.components.solaredge.const, ENERGY_DETAILS_DELAY =
timedelta(hours=1) (was 0:15:00)
2025-05-06 15:57:55.083 WARNING (MainThread)
[custom_components.constant_modifier] Patched
homeassistant.components.solaredge.const, SCAN_INTERVAL =
timedelta(hours=15) (was 0:15:00)

**Comment by thomluther** [comment]
> Wow, thanks, this seems to work, at least it is reporting the values
```text
2025-05-06 15:57:55.079 WARNING (MainThread) [custom_components.constant_modifier] Patched homeassistant.components.solaredge.const, OVERVIEW_UPDATE_DELAY = timedelta(hours=1) (was 0:15:00)
2025-05-06 15:57:55.080 WARNING (MainThread) [custom_components.constant_modifier] Patched homeassistant.components.solaredge.const, DETAILS_UPDATE_DELAY = timedelta(hours=12) (was 12:00:00)
2025-05-06 15:57:55.081 WARNING (MainThread) [custom_components.constant_modifier] Patched homeassistant.components.solaredge.const, INVENTORY_UPDATE_DELAY = timedelta(hours=12) (was 12:00:00)
2025-05-06 15:57:55.082 WARNING (MainThread) [custom_components.constant_modifier] Patched homeassistant.components.solaredge.const, POWER_FLOW_UPDATE_DELAY = timedelta(hours=1) (was 0:15:00)
2025-05-06 15:57:55.082 WARNING (MainThread) [custom_components.constant_modifier] Patched homeassistant.components.solaredge.const, ENERGY_DETAILS_DELAY = timedelta(hours=1) (was 0:15:00)
2025-05-06 15:57:55.083 WARNING (MainThread) [custom_components.constant_modifier] Patched homeassistant.components.solaredge.const, SCAN_INTERVAL = timedelta(hours=15) (was 0:15:00)
[…](#)
```
Yes, hopefully they are also appied :-)
